client/src/components/ChannelRouteSettings.tsx

```diff
@@ -110,6 +94,19 @@ class ChannelRouteSettings extends React.PureComponent<
         this.props.dispatch(storeShowOptions(this.faderIndex))
     }
 
+    getAssignedToFaderIndex = (channel: IChannelReference) => {
+        let isAssinged = -1

```

Comment:
There is a mismatch between your naming and the actual type. `isX` or `hasX` indicates that it is a boolean, but you are assigning a number to it.

That leaves me the question: Is it supposed to be a boolean or a number?

You don't have a return type on your method, so I can't check that to answer my question.
The only thing giving me a hint that the type is supposed to be a number and not a boolean is the name `getAssignedToFaderIndex` (which is a quite good name).
But since the variable `isAssigned` appears to have been misnamed and I may not be able to trust that name, then it will only be natural to also think: "Can I trust the name `getAssignedToFadeIndex`?"

These seem like trivial things, but these are the thougths/questions of a developer who sees this code for the first time.
Instead of being certain of "x variable is this y type and does z", the developer now have to stop and spend time investigating what is actually going on and this is what leads to increased time spend on maintenance.

The quickfix here would be to rename `isAssigned` to `index` or `faderIndex` and explicit state the return type and also add a return to the function.

Food for thought.

Reply:
Yes agree, and isAssinged even includes a spell error.
Have renamed it to assignedFaderIndex and added returntype